src/main/java/com/rmbcorp/javawriter/autojavac/JavaCompiler.java

```diff
@@ -0,0 +1,92 @@
+* All rights reserved.
+*
+* Permission is not granted to any person or entity to obtain a copy of this software and associated files
+* (the "Software"), to deal in the Software, which includes without limitation any rights to use, copy, modify, merge,
+* publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+* furnished to do so.  Permission may only be obtained by a signed and dated license agreement between licensor Robert
+* Brako and prospective licensee.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+* WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+* COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+* OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+package com.rmbcorp.javawriter.autojavac;
+
+import com.rmbcorp.javawriter.BuildJob;
+import com.rmbcorp.javawriter.logman.TempLogger;
+import com.rmbcorp.util.StringUtil;
+
+import java.io.*;
+
+/**
+ * Created by Aspire on 10/29/2016.
+ */
+public class JavaCompiler implements Compiler {
+
+    private final TempLogger logger;
+
+    public JavaCompiler(TempLogger logger) {
+        this.logger = logger;
+    }
+
+    @Override
+    public void compile(BuildJob javacJob) throws IOException {
+        mkdirs(javacJob.getRelativePath(), javacJob.getBinPath());
+        createFile(javacJob);
+        spawnJavac(javacJob);
+    }
+
+    private void mkdirs(String genFolder, String binFolder) {
+        boolean genCreated, binCreated;
+        File f = new File(getValidPath(genFolder));
+        genCreated = f.exists() || f.mkdir();
+        File d = new File(getValidPath(binFolder));
+        binCreated = d.exists() || d.mkdir();
+        logger.logPlain("[info]gen folder? " + genCreated + ", bin folder? " + binCreated);
+    }
+
+    private void createFile(BuildJob javacJob) throws IOException {
+        File f = new File(getValidPath(javacJob.getRelativePath()) + javacJob.getFileName() + ".java");
+        FileOutputStream fos = new FileOutputStream(f);
+        fos.write(javacJob.getFileContents().getBytes());
+        fos.close();
+    }
+
+    private String getValidPath(String relativePath) {
+        return StringUtil.isEmpty(relativePath) || StringUtil.isEmpty(relativePath.trim()) ? "" :
+                relativePath.endsWith("/") ? relativePath : relativePath + "/";
+    }
+
+    private void spawnJavac(BuildJob javacJob) throws IOException {
+        //todo: handle pause and resume for large jobs, and handle batch jobs
+        String command = "javac " + getValidCmdArg("-classpath ", javacJob.getClassPath()) + getValidCmdArg(" -d ", javacJob.getBinPath()) + " " + getValidPath(javacJob.getRelativePath()) + javacJob.getFileName() + ".java";
```

Comment:
optional: separate string build from validation. In future, think about how much validation will be done before making sys call (and using those resources), vs relying on javac for its own validation